docs/integrations/databases/convex.mdx

```diff
@@ -45,87 +73,142 @@ The next step is to configure Convex with the issuer domain provided by Clerk. F
   alt="The 'Create new template' page of the JWT Templates page in the Clerk Dashboard. There is a red box surrounding the 'Issuer' section."
 />
 
-In your terminal, run the following command from your Convex app directory:
+In your `convex` folder, add an `auth.config.js` file with the following configuration:
+
+```ts title="convex/auth.config.js"
+export default {
+  providers: [
+    {
+      domain: "https://your-issuer-url.clerk.accounts.dev/",
+      applicationID: "convex",
+    },
+  ]
+};
+```
+
+Replace the `domain` string with the **Issuer** URL you copied.
+
+### Deploy your changes to Convex
+
+Run `npx convex dev` to automatically sync your configuration to your backend.
+
+### Install `@clerk/clerk-react`
+
+Install Clerk's React SDK by running the following command in your terminal:
+
+<CodeBlockTabs type="installer" options={["npm", "yarn",  "pnpm" ]}>
+  ```bash filename="terminal"
+  npm install @clerk/clerk-react
+  ```
+  ```bash filename="terminal"
+  yarn add @clerk/clerk-react
+  ```
+  
+  ```bash filename="terminal"
+  pnpm add @clerk/clerk-react
+  ```
+
+</CodeBlockTabs>
+
+### Set environment keys
+
+In your React project's root folder, you may have an `.env` file alongside `package.json` and other configuration files. If you don't see it, create it.
+
+Add the following code to your `.env` file to set your public key.
 
-```sh filename="terminal"
-npx convex auth add
+**Pro tip!** If you are signed into your [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=api-keys), you can copy your publishable key below.
+
+<InjectKeys>
+
+```sh filename=".env"
+VITE_CLERK_PUBLISHABLE_KEY={{pub_key}}
 ```
 
-Paste the **Issuer** URL you copied when it prompts you for the identity provider's URL.
+</InjectKeys>
+
+### Import the Publishable Key
+
+You will need to import your Publishable Key into your application. You can add an `if` statement to check that it is imported and that it exists. This will prevent running the application without the Publishable Key, and will also prevent TypeScript errors.
+
+```tsx filename="src/main.tsx" {6-7, 9-11}
+import React from 'react'
+import ReactDOM from 'react-dom/client'
+import App from './App.tsx'
+import './index.css'
 
-The application/client ID will need to match the `aud` claim (default is `convex`).
+// Import your publishable key
+const PUBLISHABLE_KEY = import.meta.env.VITE_CLERK_PUBLISHABLE_KEY
 
-```sh
-? Enter the identity providers Domain URL: https://your-issuer-url.clerk.accounts.dev/
-? Enter your application/client ID with this identity provider: convex
-? Would you like to add another provider? No
-? Configuration updated. Run npx convex dev or npx convex deploy to sync these changes.
+if (!PUBLISHABLE_KEY) {
+  throw new Error("Missing Publishable Key")
+}
+
+ReactDOM.createRoot(document.getElementById('root')!).render(
+  <React.StrictMode>
+      <App />
+  </React.StrictMode>,
+)
```

Comment:
I find having this whole step (from line 129) with code that doesn't match the onboarding flow, confusing. The code here is included in the next step. Can we remove it? We can leave the throwing (although I'm pretty sure ClerkProvider will throw anyway)

Reply:
Great point ! I copied and pasted this from the React quickstart in a haste, and you are completely right that this step is misplaced. I have removed it :) Great catch.